### PR TITLE
pypy3: migrate to Python 3.8

### PR DESCRIPTION
**Pull request by equal-l2**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

From PyPy 7.3.8, the Python 3.8 implementation by PyPy is considered as stable.
https://www.pypy.org/posts/2022/02/pypy-v738-release.html

### CODE REVIEW

File: Formula/pypy3.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-  url "https://downloads.python.org/pypy/pypy3.7-v7.3.9-src.tar.bz2"
-  sha256 "70426163b194ee46009986eea6d9426098a3ffb552d9cdbd3dfaa64a47373f49"
   license "MIT"
-  head "https://foss.heptapod.net/pypy/pypy", using: :hg, branch: "py3.7"
+  head "https://foss.heptapod.net/pypy/pypy", using: :hg, branch: "py3.8"
+
+  stable do
+    # M1 support is planned to be included in the next release
+    # See https://www.pypy.org/posts/2022/07/m1-support-for-pypy.html
+    depends_on arch: :x86_64
```
Comment (carlocab) [inline Formula/pypy3.rb:10]: Will they release a version of `pypy` with M1 support as well? Or is there a way of bootstrapping this without `pypy`?

Comment (equal-l2) [reply Formula/pypy3.rb:10]: PyPy seems to share the base code for all Python versions (3.x and 2.7), so there will likely be the M1 build for `pypy` on the next release.

**Comment by carlocab** [comment]
The `test` needs updating.

**Comment by equal-l2** [comment]
I have no clew about the failure.
Do you have any thought, @bayandin, @Bo98 ?
(These people handled https://github.com/Homebrew/homebrew-core/pull/99892)

**Comment by equal-l2** [comment]
After several retries, I can reproduce the test failure but still have no clue how to resolve the problem.
List of the oddities I saw:
- `/usr/local/share/pypy3` is empty (should contains `pip` at least)
- `/usr/local/lib/pypy3/site-packages` only contains a broken symlink `site-package`, which points to `../../../../lib/pypy3/site-packages`

**Comment by equal-l2** [comment]
I'm closing this for now because I see no actionable way to resolve this.

**Pull request closed without merging**